app/models/daily.rb

```diff
@@ -6,4 +6,22 @@ class Daily < ApplicationRecord
   has_many :scores, dependent: :destroy
   has_one :supporter, dependent: :destroy
 
+
+  def weeks_scores
+    results = self.scores.created_between((Date.today - 7), Date.today).order(date: :desc).pluck(:score)
+
+    while results.length < 7
+      results.unshift(nil)
+    end
+    results
+  end
+
+  def weeks_average
+    results = self.scores.created_between((Date.today - 7), Date.today).order(date: :desc).pluck(:score)
+    results.reduce(:+)/results.length unless results.empty?
+  end
+
+  def table_row
+    [[goal] + weeks_scores.map {|n| n || 0}]
+  end
 end
```

Comment:
@case-eee I built the table_row method to pass data in the format my chart requires. Since it's partially a formatting issue, should this be done somewhere other than the model, like in a helper method?